plugins/backend/catalog-backend-module-aws/src/providers/AWSDynamoDbTableDataProvider.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ANNOTATION_LOCATION,
+  ANNOTATION_ORIGIN_LOCATION,
+} from '@backstage/catalog-model';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-backend';
+import { fromTemporaryCredentials } from '@aws-sdk/credential-providers';
+import { STS } from '@aws-sdk/client-sts';
+import { Config } from '@backstage/config';
+
+import { DynamoDB, DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb';
+import { merge } from 'lodash';
+import { mapColumnsToEntityValues } from '../utils/columnMapper';
+import * as winston from 'winston';
+
+export type ValueMapping = {
+  entityPath: string;
+  template?: string;
+};
+
+type DdbTableDataConfigOptions = {
+  tableName: string;
+  identifierColumn?: string;
+  columnValueMapping?: { [key: string]: ValueMapping };
+};
+
+/**
+ * Provides entities from AWS DynamoDB service.
+ */
+export class AWSDynamoDbTableDataProvider implements EntityProvider {
+  private readonly accountId: string;
+  private readonly roleArn: string;
+  private readonly tableDataConfig: DdbTableDataConfigOptions;
+  private readonly logger: winston.Logger;
+
+  private connection?: EntityProviderConnection;
+
+  static fromConfig(config: Config, options: { logger: winston.Logger }) {
+    return new AWSDynamoDbTableDataProvider(
+      config.getString('accountId'),
+      config.getString('roleArn'),
+      config.get<DdbTableDataConfigOptions>('dynamodbTableData'),
+      options.logger,
+    );
+  }
+
+  constructor(
+    accountId: string,
+    roleArn: string,
+    options: DdbTableDataConfigOptions,
+    logger: winston.Logger,
+  ) {
+    this.accountId = accountId;
+    this.roleArn = roleArn;
+    this.tableDataConfig = options;
+    this.logger = logger;
+  }
+
+  getProviderName(): string {
+    return `aws-dynamo-db-table-${this.accountId}-${this.tableDataConfig.tableName}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+  }
+
+  async run(): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const credentials = fromTemporaryCredentials({
+      params: { RoleArn: this.roleArn },
+    });
+
+    const ddb = DynamoDBDocument.from(
```

Comment:
What's the difference between this client and the one below?

Reply:
Honestly just curious

Reply:
Easier to handle with this wrapper. The DynamoDBDocument handles a lot of the serializing needed when talking with DDB. On line `117` you can see that it returns pure rows only.